src/gui/ByteWidget.cpp

```diff
@@ -0,0 +1,48 @@
+﻿#include "ByteWidget.hpp"
+
+#include <QGridLayout>
+#include <QLabel>
+#include <QSpinBox>
+
+ByteWidget::ByteWidget(QString description, quint8 defaultValue,
+                       QWidget *parent) :
+    QWidget(parent),
+    m_description(std::move(description)), m_defaultValue(defaultValue)
+{
+    init();
+}
+
+void ByteWidget::init()
+{
+    m_value = m_defaultValue;
+
+    auto layout    = new QGridLayout(this);
+    m_valueSpinBox = new QSpinBox(this);
+    m_valueSpinBox->setValue(m_value);
+    m_valueSpinBox->setMinimum(0);
+    m_valueSpinBox->setMaximum(255);
+    connect(m_valueSpinBox, QOverload<int>::of(&QSpinBox::valueChanged),
+            [=](int i) {
+                m_value = i;
+                update();
+            });
+    m_decValueLabel = new QLabel(this);
+    m_hexValueLabel = new QLabel(this);
+    int index       = 0;
+    layout->addWidget(m_valueSpinBox, 0, index++);
+    layout->addWidget(m_decValueLabel, 0, index++);
+    layout->addWidget(m_hexValueLabel, 0, index++);
+    check();
+}
+
+void ByteWidget::check()
```

Comment:
What does it check???

Reply:
I used the same methods used in LedWidget